cloud/scope/vpc_cluster.go

```diff
@@ -686,3 +742,321 @@ func (s *VPCClusterScope) buildCOSObjectHRef() (*string, error) {
 	s.V(3).Info("building image ref", "href", href)
 	return ptr.To(href), nil
 }
+
+// ReconcileSubnets reconciles the VPC Subnet(s).
+// For Subnets, we collect all of the required subnets, for each Plane, and reconcile them individually. Requeing if one is missing or just created. Reconciliation is attempted on all subnets each loop, to prevent single subnet creation per reconciliation loop.
+func (s *VPCClusterScope) ReconcileSubnets() (bool, error) {
+	var subnets []infrav1beta2.Subnet
+	var err error
+	// If no ControlPlane Subnets were supplied, we default to create one in each availability zone of the region.
+	if s.IBMVPCCluster.Spec.Network.ControlPlaneSubnets == nil || len(s.IBMVPCCluster.Spec.Network.ControlPlaneSubnets) == 0 {
+		subnets, err = s.buildSubnetsForZones()
+		if err != nil {
+			return false, fmt.Errorf("error failed building control plane subnets: %w", err)
+		}
+	} else {
+		subnets = s.IBMVPCCluster.Spec.Network.ControlPlaneSubnets
+	}
+
+	// Reconcile Control Plane subnets.
+	requeue := false
+	for _, subnet := range subnets {
```

Comment:
If no inter dependency, In future for performance optimization we can think of reconciling data and control plane subnets concurrently.

Reply:
That is likely possible, if subnet creation is found to take longer than desired I can revisit that. I know Custom Image and LB's are the larger time sinks (in terms of waiting for them to be `Ready`, and SG/SGR reconciliation is heavily complex and can take time as well.